src/app/home/menu/menu.component.ts

```diff
@@ -0,0 +1,61 @@
+import { CommonModule } from '@angular/common';
+import { Component } from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { MatSnackBar, MatSnackBarModule, MatSnackBarRef, SimpleSnackBar } from '@angular/material/snack-bar';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { NavigationEnd, Router } from '@angular/router';
+import { filter, map, Observable } from 'rxjs';
+import { NewMessageDialogComponent } from 'src/app/guest-book/components/new-message-dialog/new-message-dialog.component';
+
+@Component({
+  selector: 'app-menu',
+  templateUrl: './menu.component.html',
+  styleUrls: ['./menu.component.scss'],
+  standalone: true,
+  imports: [MatToolbarModule, MatButtonModule, CommonModule, MatDialogModule, MatSnackBarModule]
+})
+export class MenuComponent {
+  isGuestBookRoute$!: Observable<boolean>;
+  constructor(
+    private dialog: MatDialog,
+    private snackBar: MatSnackBar,
+    private router: Router
+  ) { }
+
+  ngOnInit(): void {
+    this.isGuestBookRoute$ = this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      map(() => this.router.url.includes('/guest-book'))
+    );
+  }
+
+  onGuestBookClick() {
+    this.router.navigate(['/guest-book']);
```

Comment:
usually it's cleaner to user routerlink - instead of handling navigation manually
https://angular.dev/api/router/RouterLink?tab=usage-notes